src/sentry/static/sentry/app/views/dashboards/exploreWidget.jsx

```diff
@@ -44,51 +45,63 @@ class ExploreWidget extends React.Component {
 
     return (
       <DropdownMenu>
-        {({isOpen, getRootProps, getActorProps, getMenuProps}) => {
-          return (
-            <ExploreRoot {...getRootProps()} isOpen={isOpen}>
-              <div {...getActorProps()}>
-                <ExploreButton isOpen={isOpen}>
-                  {t('Explore Data')}
-                  <Chevron isOpen={isOpen} src="icon-chevron-right" />
-                </ExploreButton>
-              </div>
-              <ExploreMenu {...getMenuProps({isOpen})}>
-                {discoverQueries.map(query => (
-                  <ExploreRow key={query.name}>
-                    <QueryName>{query.name}</QueryName>
-
-                    <Feature features={['discover']} organization={organization}>
-                      {({hasFeature}) => (
+        {({isOpen, getRootProps, getActorProps, getMenuProps}) => (
+          <ExploreRoot {...getRootProps()} isOpen={isOpen}>
+            <div {...getActorProps()}>
+              <ExploreButton isOpen={isOpen}>
+                {t('Explore Data')}
+                <Chevron isOpen={isOpen} src="icon-chevron-right" />
+              </ExploreButton>
+            </div>
+            <ExploreMenu {...getMenuProps({isOpen})}>
+              {discoverQueries.map(query => (
+                <ExploreRow key={query.name}>
+                  <QueryName>{query.name}</QueryName>
+
+                  <Feature features={['discover']} organization={organization}>
```

Comment:
@leedongwei This is great. I think we want to hide the Discover v1 link if v2 is accessible. Maybe use this implementation to check if either V2 or legacy is preferred:  https://github.com/getsentry/sentry/blob/8d31f8651558b3f9a5f65dc45e0f439c5ac19d55/src/sentry/static/sentry/app/components/sidebar/index.jsx#L230-L278 ?

Reply:
Users could have both discover + discover-query and I think we'd only want to show one button, preferring to sync with their preference and biasing to discover2.

Reply:
Given that Discover1 will be deprecated soon-ish, I don't want to refactor this function out into a file to be shared, so I copied/pasted this function from `sidebar/index.jsx`.

Reply:
Yea that's fine.